internal/storage/pmem/pmem.go

```diff
@@ -36,18 +60,54 @@ func Mount(ctx context.Context, device uint32, target string) (err error) {
 		trace.Int64Attribute("device", int64(device)),
 		trace.StringAttribute("target", target))
 
-	if err := osMkdirAll(target, 0700); err != nil {
+	source := fmt.Sprintf(pMemFmt, device)
+	return mountInternal(source, target)
+}
+
+// MountDM creates dm-linear block device and mounts it on `target`
+func MountDM(ctx context.Context, device uint32, deviceStart, deviceSize int64, target string) (err error) {
+	_, span := trace.StartSpan(ctx, "pmem::MountDM")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	lt := dm.PMemLinearTarget(deviceSize, fmt.Sprintf(pMemFmt, device), deviceStart)
+	linearDeviceName := fmt.Sprintf(linearDeviceFmt, device, deviceStart, deviceSize)
+
+	span.AddAttributes(
+		trace.Int64Attribute("device", int64(device)),
+		trace.Int64Attribute("deviceStart", deviceStart),
+		trace.Int64Attribute("sectorSize", deviceSize),
+		trace.StringAttribute("target", target),
+		trace.StringAttribute("table", fmt.Sprintf("%s: '%d %d %s'", linearDeviceName, lt.SectorStart, lt.Length, lt.Params)))
+
+	var dmpath string
```

Comment:
Does this actually need to be declared, as we just create it one line down anyways? There's no path where it gets assigned to something in a conditional